snowflake/cortex/_embed_text_1024.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional, Union, List
+
+from snowflake import snowpark
+from snowflake.cortex._util import (
+    CORTEX_FUNCTIONS_TELEMETRY_PROJECT,
+    SnowflakeConfigurationException,
+    call_sql_function,
+)
+from snowflake.ml._internal import telemetry
+
+
+SUPPORTED_MODELS: List[str] = [
+    "nv-embed-qa-4",
+    "multilingual-e5-large",
+]
+
+
+@telemetry.send_api_usage_telemetry(
+    project=CORTEX_FUNCTIONS_TELEMETRY_PROJECT,
+)
+def EmbedText1024(
+    model: Union[str, snowpark.Column],
+    text: Union[str, snowpark.Column],
+    session: Optional[snowpark.Session] = None,
+) -> Union[str, snowpark.Column]:
+    """TextEmbed calls into the LLM inference service to embed the text.
+
+    Args:
+        model: A Column of strings representing the model to use for embedding. The value
+               of the strings must be within the SUPPORTED_MODELS list.
+        text: A Column of strings representing input text.
+        session: The snowpark session to use. Will be inferred by context if not specified.
+
+    Returns:
+        A column of vectors containing embeddings.
+    """
+
+    if model not in SUPPORTED_MODELS:
```

Comment:
As the models will be changing over time, we want to have the check for if a model is supported on the server side.

When this happens, we should throw an exception as-is. Is that sufficient?

Reply:
Yep, that works for me. Updating the code to match.